### PR TITLE
NoBlankLines*: fix removing lines consisting only of spaces

### PR DESCRIPTION
**Pull request by kubawerlos**
We have 3 `NoBlankLines*` fixers and they behave differently for lines consisting only of spaces:
- `NoBlankLinesAfterClassOpeningFixer` - removes line of spaces only when next line is indented
- `NoBlankLinesAfterPhpdocFixer` - same as above
- `NoBlankLinesBeforeNamespaceFixer` - removes indentation of the next line

What would be the proper behaviour (apart of removing indentation of next line what is obviously a bug) - to keep lines consisting only of spaces or to remove them?

_Tests are to demonstrate the problem, obviously have to be fixed as well._

### CODE REVIEW

**Comment by kubawerlos** [comment]
By making a proxy for `2.12` and deprecation for `master`?

**Comment by dmvdbrugge** [comment]
> how about merging those into `no_extra_blank_lines` ?

There's a difference between "no extra" (meaning at most 1) or "no" (meaning 0).

I'd be for combining the ones pointed out into a single fixer (just like the `blank_line_before_statement`), however think that it should be a new one. (With the deprectations ofc.)

**Comment by SpacePossum** [comment]
if I expect 0 than 1 is `extra` to me

**Comment by dmvdbrugge** [comment]
But how are you going to configure where you expect zero and where one? Sounds like configuration hell. I for one, wouldn't want `NoBlankLinesBeforeNamespaceFixer` while I do want the others.

**Comment by kubawerlos** [comment]
Okay, as there obviously going to be discussion about new fixer to remove blank lines let's focus here about fixing the bugs only.

File: tests/Fixtures/Integration/priority/no_whitespace_in_blank_line,no_blank_lines_after_phpdoc.test [hunk 1]
```diff
@@ -1,22 +0,0 @@
---TEST--
```
Comment (keradus) [inline tests/Fixtures/Integration/priority/no_whitespace_in_blank_line,no_blank_lines_after_phpdoc.test]: please do not remove tests.
ever

**Comment by keradus** [comment]
Thank you @kubawerlos.

File: tests/Fixtures/Integration/priority/no_whitespace_in_blank_line,no_blank_lines_after_phpdoc.test (hunk 1 above)
Comment (kubawerlos) [reply tests/Fixtures/Integration/priority/no_whitespace_in_blank_line,no_blank_lines_after_phpdoc.test]: Oh, yeah, I haven't thought about moving it from `priority` to `misc`. Next time I will remember.

**Pull request merged**